### ISSUE TITLE
file extention consistency?

### ISSUE THREAD

**Issue by techdragon** [getnikola/nikola#2125]
There are `*.txt` files full of restructured text. 
Both Mako and Jinja2 templates are in files just called `*.tmlp`. 

This makes working with Nikola much more difficult than it should be in both PyCharm and Sublime Text.
Is there a reason they are named this way? 
Or is it historical and we could move them to `*.rst`, `*.html` or `*.mako` and `*.jinja2` without causing problems?

**Comment by ralsina** [comment]
.txt is the extension recommended by [the docutils team](http://docutils.sourceforge.net/FAQ.html#what-s-the-standard-filename-extension-for-a-restructuredtext-file)

As for mako and jinja, well, it's random and has no reason behind it, really. However, the template filenames are all around the code  so renaming them is not exactly easy.

**Comment by Aeyoun** [comment]
I’d score `.mako` and `.jinja2` a bit higher on usability as our technical users wanting to customize would see the name of the tempting engine to put into their web searches effortlessly and would find more relevant information.

Instead of hard-coding `.tmpl` everywhere, we could send the template name to a utility function that would match the file extension to the template engine OR `.tmpl` if a file with that name is found in the theme (backwards compatibility)

**Comment by Kwpolska** [comment]
> Renaming .mako to .jinja2 in jinjify.py would be a three liner.

This is the least of our worries.  The actual problem would be convincing Mako and Jinja2 to translate .tmpl into .mako/.jinja2 — and those names are embedded in existing templates (and we all know how this ends after #2011…)

**Comment by ralsina** [comment]
I am inclined to not do this.

**Comment by Kwpolska** [comment]
So, that's a wontfix.
